### ISSUE TITLE
[Feature Request]   Add ability to edit files i.e. with NVIM 

### ISSUE THREAD

**Issue by dwilliam62** [juftin/browsr#42]
I really like the look and remote capabilities.  It may not be possible for remote connections, but for local files the ability to browse and do quick edits would be amazing

**Comment by juftin** [comment]
This is coming! (eventually). I've had my eye on this feature for a while - that's why I recently refactored browsr to be more modular.

It wouldn't be NVIM - instead it's a [textual](https://github.com/textualize/textual/) (the underlying framework behind browsr) widget called [TextArea](https://textual.textualize.io/widget_gallery/#textarea). 

It will improve the code viewing experience too as it supports things like text highlighting. It will even support (some) remote filesystems. 

I'm specifically waiting on more languages to be supported by syntax highlighting for this (https://github.com/Textualize/textual/pull/4160). 

As far as the UX is concerned I was thinking about an `E` keyboard shortcut that launches you into editing mode in a new window. It'll probably be more similar to `nano` than `vim` - since that's what I know and I'd like it to be pretty simple.

**Comment by dwilliam62** [comment]
Sounds good.   However many people prefer their own editors   NVIM/EMACS
because they are set up like mine for language parsing    I would suggest
making that an option.

Regards,
Don

**Comment by juftin** [comment]
Fair enough, but figuring out how to implement NVIM/EMACS inside a Python process run by [textual](https://github.com/textualize/textual/) will likely be prohibitively difficult. I typically have to work with existing "widgets" that are already in the textual ecosystem, like the built-in `TextArea`.

Invoking NVIM/EMACs via a shell plugin (https://github.com/juftin/browsr/issues/44) is probably a cleaner path for this.

**Comment by dwilliam62** [comment]
Yeah this is what I was imagining. It would call $EDITOR and pass the
filename   Or via a settings option.    I was not expecting you to do
something within this framework.

**Comment by davep** [comment]
> Fair enough, but figuring out how to implement NVIM/EMACS inside a Python process run by [textual](https://github.com/textualize/textual/) will likely be prohibitively difficult.

@juftin Apologies for barging in, and perhaps I've misread this, but it should be [pretty straightforward](https://textual.textualize.io/guide/app/#suspend-context-manager).

**Comment by juftin** [comment]
> @juftin Apologies for barging in, and perhaps I've misread this, but it should be [pretty straightforward](https://textual.textualize.io/guide/app/#suspend-context-manager).

Thank you Dave - no need to apologies! I love having your textual knowledge on stuff like this. I actually stumbled on the concept of suspending the context manager after looking into this - this is an awesome solution to this.  Ideally I'd be able to do this with the `TextArea` widget though so editing files would be supported on both local and remote filesystems (like editing an S3 file). Whether it's used for editing files or not, there is also a need to be able to run a local terminal session alongside `browsr` - I'm initially more inclined to use something like [textual-terminal](https://github.com/mitosch/textual-terminal) like this though.

**Comment by davep** [comment]
Sure thing, I thought it worth a mention just in case. Should be useful for those who prefer `$EDITOR` support (and as I'm sure you know, most who would likely have their editor set to support remote editing, so that might be worth keeping in mind -- see `tramp` in Emacs as an example).
